src/backend/columnar/columnar_storage.c

```diff
@@ -354,8 +356,7 @@ ColumnarStorageGetReservedOffset(Relation rel, bool force)
 bool
 ColumnarStorageIsCurrent(Relation rel)
 {
-	RelationOpenSmgr(rel);
-	BlockNumber nblocks = smgrnblocks(rel->rd_smgr, MAIN_FORKNUM);
+	BlockNumber nblocks = smgrnblocks(RelationGetSmgr(rel), MAIN_FORKNUM);
```

Comment:
These are problematic before 15, FWIW.

Reply:
Hmm, I thought if you opened right before using the `rel->rd_smgr`, it was fine. To be a problem doesn't there need to be some other call in between the `RelationOpenSmgr()` and the `smgr*()` call that uses it?